homeassistant/components/modbus/binary_sensor.py

```diff
@@ -138,6 +139,9 @@ def __init__(
         idx += 1
         self._attr_name = f"{entry[CONF_NAME]} {idx}"
         self._attr_device_class = entry.get(CONF_DEVICE_CLASS)
+        self._attr_unique_id = entry.get(CONF_UNIQUE_ID)
+        if self._attr_unique_id:
```

Comment:
The modbus integration have 100% test coverage, so please write tests to show that your code works.

Reply:
Can you point me at a similar test I can base this one please?

Reply:
Look tests/components/modbus